WORKSPACE

```diff
@@ -0,0 +1,107 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+# Download Extra java rules
+http_archive(
+    name = "rules_jvm_external",
+    sha256 = "08ea921df02ffe9924123b0686dc04fd0ff875710bfadb7ad42badb931b0fd50",
+    strip_prefix = "rules_jvm_external-6.1",
+    url = "https://github.com/bazelbuild/rules_jvm_external/releases/download/6.1/rules_jvm_external-6.1.tar.gz",
+)
+
+load("@rules_jvm_external//:repositories.bzl", "rules_jvm_external_deps")
+
+rules_jvm_external_deps()
+
+load("@rules_jvm_external//:defs.bzl", "maven_install")
+
+maven_artifacts = [
+    "org.ejml:ejml-simple:0.43.1",
+    "com.fasterxml.jackson.core:jackson-annotations:2.15.2",
+    "com.fasterxml.jackson.core:jackson-core:2.15.2",
+    "com.fasterxml.jackson.core:jackson-databind:2.15.2",
+    "us.hebi.quickbuf:quickbuf-runtime:1.3.3",
+    "com.google.code.gson:gson:2.10.1",
+]
+
+maven_install(
+    name = "maven",
+    artifacts = maven_artifacts,
+    repositories = [
+        "https://repo1.maven.org/maven2",
+        "https://frcmaven.wpi.edu/artifactory/release/",
+    ],
+)
+
+# Download toolchains
+http_archive(
+    name = "rules_bzlmodrio_toolchains",
+    sha256 = "cd3ff046427e9c6dbc0c86a458c8cf081b8045fc3fb4265d08c0ebfc17f9cb30",
+    url = "https://github.com/bzlmodRio/rules_bzlmodRio_toolchains/releases/download/2024-1/rules_bzlmodRio_toolchains-2024-1.tar.gz",
```

Comment:
How does WPILib want to manage the third-party dependencies being imported here to support toolchains and bazel stuff?

Also, looking at this repo I don't see any license?

Reply:
All the `bzlmodRio` stuff is mine. I can add whatever license is desirable.

Reply:
I take it that the bzlmodRio stuff (and pulling it in via http_archive) is akin to our current Gradle plugins?  We'd want to move/fork the relevant repos into wpilibsuite for long term use.  I know this PR is focused on allwpilib, where we can pretty much assume anyone building it is online, but how does this work in the team build case with no internet?  Can the http_archive stuff be cached offline?

Reply:
Getting stuff distributed with the installer would be a bit unusual for a bazel project, but should definitely be feasible. Normally the http_archive things being downloaded get saved into a cache on the user's machine in a reasonably standard spot based on the `sha256` of the `http_archive`. I imagine there are a couple of different ways that you can manage offline distribution, with varying levels of hackiness.

Reply:
So there would be two options.:
- As @jkuszmaul you could place the files in the magic directory that bazel knows about
- `http_archive` supports multiple mirrors, and I just tested that you could have a mirror that is a  `file://`. So that could point to the magic directory offline installers put stuff

Reply:
https://github.com/bazelbuild/bazel/issues/18934 is the official upstream feature number and the corresponding docs for how they expect it to work.